CodingAChat/Scenes/Login/LoginCoordinator.swift

```diff
@@ -40,6 +40,7 @@ extension LoginCoordinator: LoginViewControllerDelegate {
                 self?.onLogin?(result.user)
                 
             } else if let error = error {
+                UIAlertController.show(message: "Wrong email or password", from: self!.loginViewController)
```

Comment:
Same here, you could add let self = self in if condition and safety remove self!